### ISSUE TITLE
How to require a specific build flavour in a dependency in a stable project

### ISSUE THREAD

**Issue by schmunk42** [composer/composer#6489]
### TL;dr

In a nutshell, how can I require a specific release of ckeditor in a package? Without having to be on `stability: dev` or specifying the dependency in my root `composer.json` (Bonus without: Having multiple repos)

My `composer.json`:

```json
{
  "repositories": [
      {
          "type": "vcs",
          "url": "https://github.com/schmunk42/ckeditor-releases.git"
      }
  ],
    "require": {
        "ckeditor/ckeditor": "^4.6+basic"
    }
}
```

Output of `composer diagnose`:

```
> composer diagnose        
Checking composer.json: WARNING
No license specified, it is recommended to do so. For closed-source software you may use "proprietary" as license.
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com oauth access: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
```

When I run this command:

```
composer update
```

Assuming that I've changed my "build-flavor"

I get the following output:

```
 composer update                    
Loading composer repositories with package information
Updating dependencies (including require-dev)                
Nothing to install or update
Generating autoload files
```

And I expected this to happen:

Get a specifc flavour "full", "basic", "standard"

Underlying issue is described in:
- https://github.com/ckeditor/ckeditor-releases/issues/41
- https://github.com/2amigos/yii2-ckeditor-widget/issues/106
- https://github.com/2amigos/yii2-ckeditor-widget/pull/96

**Comment by alcohol** [comment]
Look at https://packagist.org/packages/ckeditor/ckeditor (versions list).

Try the following as a constraint: `dev-basic/4.6.x`.

**Comment by alcohol** [comment]
But eh, long story short, their versioning schema is not semver compatible and thus not supported. Composer attempts a best effort, but we cannot support everything. So, tough luck unfortunately :-)

**Comment by schmunk42** [comment]
> But eh, long story short, their versioning schema is not semver compatible and thus not supported.

Yeah, noticed that also ;) https://github.com/ckeditor/ckeditor-releases/issues/41#issuecomment-307933384

What would be a correct versioning schema for this? How would they have to tag their versions to be compliant to composer?

- `4.7.0+full` does not work because meta-data is ignored
- `4.7.0-full` is not a valid tag
- `dev/4.7.0-full` does not work in `stable` projects

**Comment by alcohol** [comment]
Honestly? It should be released as 3 different packages. E.g. `ckeditor/ckeditor` (standard), `ckeditor/ckeditor-basic`, `ckeditor/ckeditor-full`.

**Comment by schmunk42** [comment]
> Honestly? It should be released as 3 different packages.

It's a valid use-case. But I can also understand why this is not supported from the composer side.
